src/graph/topo.cc

```diff
@@ -1036,6 +1041,73 @@ ncclResult_t scclGetAllAlgoFromXMLFilesAndSetComm(struct ncclComm* comm, const c
   return ncclSuccess;
 }
 
+ncclResult_t scclGetAllAlgoFromSCCLConfigAndSetComm(struct ncclComm* comm, const char* str){
+  INFO(NCCL_INIT, "SCCL: Parsing config %s", str);
+  struct ncclXml* xml;
+
+  comm->scclRegistrations = NULL;
+  comm->nScclRegistrations = 0;
+
+  NCCLCHECK(ncclCalloc(&xml, 1));
+  NCCLCHECK(scclGetXmlConfigFromFile(str, xml));
+
+  struct ncclXmlNode* topNode;
+  NCCLCHECK(xmlFindTag(xml, "sccl_algos", &topNode));
+
+  for (int s=0; s < topNode->nSubs; s++) {
+    struct ncclXmlNode* node = topNode->subs[s];
+    if (strcmp(node->name, "load") == 0) {
+      if (comm->numberOfSCCAlgorithms == SCCL_MAX_NUM_ALGOS){
+        WARN("SCCL: too many algorithms (%d) specified in environment variable SCCL_XML_FILES. The rest will be ignored.", comm->numberOfSCCAlgorithms);
+        break;
+      }
+
+      const char *path;
+      NCCLCHECK(xmlGetAttrStr(node, "path", &path));
+
+      int hasMinsize = false;
+      NCCLCHECK(xmlAttrExists(node, "minsize", &hasMinsize));
```

Comment:
are these in bytes or counts? can we call them minBytes to be sure?

Reply:
They are bytes, so renaming is a good idea.